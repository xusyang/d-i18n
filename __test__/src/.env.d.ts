/// <reference types="vite/client" />

interface ImportMetaEnv {
  readonly VITE_APP_ENV: string
  readonly VITE_APP_BASE_API: string
  readonly VITE_APP_LOWCODE_ENTRY: string
  readonly VITE_BUILD_COMPRESS: string
}

interface ImportMeta {
  readonly env: ImportMetaEnv
}
