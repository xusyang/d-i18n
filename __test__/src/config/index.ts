export default {
  BASE_API: import.meta.env.VITE_APP_BASE_API,
  LOWCODE_ENTRY: import.meta.env.VITE_APP_LOWCODE_ENTRY as string,
  LOWCODE_PREFIX: import.meta.env.VITE_APP_LOWCODE_PREFIX as string,
  FLOW_ENTRY: import.meta.env.VITE_APP_FLOW_ENTRY as string,
  FLOW_PREFIX: import.meta.env.VITE_APP_FLOW_PREFIX as string,
}
