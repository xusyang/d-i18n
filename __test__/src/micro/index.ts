import { start, registerMicroApps } from 'qiankun'
import NProgress from 'nprogress'

import apps from './apps'

/**
 * 开启微前端
 */
export const startMicroApp = function () {
  console.log('register micro app')
  registerMicroApps(apps, {
    beforeLoad: () => {
      console.log('beforeLoad')
      NProgress.start()
      return Promise.resolve()
    },

    afterMount: () => {
      console.log('afterMount')
      NProgress.done()
      return Promise.resolve()
    },
  })
}
