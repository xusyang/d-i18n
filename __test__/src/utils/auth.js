import Cookies from 'js-cookie'

const TokenKey = 'CAPA-Admin-Token'
const LegalKey = 'CAPA-LEGAL'
const removeKeys = [TokenKey, LegalKey]
const ExpiresInKey = 'Admin-Expires-In'

export function getToken() {
  return Cookies.get(TokenKey)
}

export function setToken(token) {
  return Cookies.set(TokenKey, token)
}

export function getLegal() {
  return Cookies.get(LegalKey)
}

export function setLegal(id) {
  return Cookies.set(LegalKey, id)
}

export function removeToken() {
  removeKeys.forEach((key) => {
    Cookies.remove(key)
  })
}

export function getExpiresIn() {
  return Cookies.get(ExpiresInKey) || -1
}

export function setExpiresIn(time) {
  return Cookies.set(ExpiresInKey, time)
}

export function removeExpiresIn() {
  return Cookies.remove(ExpiresInKey)
}
