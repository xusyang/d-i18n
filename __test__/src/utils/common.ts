export function normalizeTree(parentId: number, rawData: any[]) {
  const root = rawData.filter((x: any) => x.parentId === parentId)
  root.forEach((item: any) => {
    item.children = normalizeTree(item.id, rawData)
    if (item.children.length <= 0) delete item.children
  })
  return root
}
