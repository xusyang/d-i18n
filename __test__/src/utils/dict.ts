import { ref, toRefs } from 'vue'
import { getDicts } from '@/api/system/dict/data'

/**
 * 模型的操作类型
 */
export const MODEL_OPERATOR_TYPE = {
  新增: 1,
  编辑: 2,
}

export const DEPT_TYPE = {
  前台: 1,
  中台: 2,
  后台: 3,
}

export const DICT_KEYS = {
  DEPT_TYPE: 'DEPT_TYPE',
  MODEL_OPERATOR_TYPE: 'MODEL_OPERATOR_TYPE',
}

/**
 * 获取字典数据
 */
export function useDict(...args) {
  // 是否需要做缓存
  const res = ref({})
  return (() => {
    args.forEach((d) => {
      res.value[d] = []
      getDicts(d).then((resp) => {
        res.value[d] = resp.data.map((p) => ({ label: p.dictLabel, value: p.dictValue, elTagType: p.listClass }))
      })
    })
    return toRefs(res.value)
  })()
}
