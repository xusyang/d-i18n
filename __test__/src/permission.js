import router from './router'
import store from './store'
import { ElMessage } from 'element-plus'
import NProgress from 'nprogress'
import 'nprogress/nprogress.css'
import { getToken } from '@/utils/auth'
import { isHttp } from '@/utils/validate'
import { isRelogin } from '@/utils/request'

NProgress.configure({ showSpinner: false })

const whiteList = ['/login', '/auth-redirect', '/bind', '/register']

router.beforeEach(async (to, from, next) => {
  NProgress.start()
  if (getToken()) {
    to.meta.title && store.dispatch('settings/setTitle', to.meta.title)
    /* has token*/
    if (to.path === '/login') {
      next({ path: '/' })
      NProgress.done()
    } else {
      if (store.getters.roles.length === 0) {
        isRelogin.show = true
        // 判断当前用户是否已拉取完user_info信息

        try {
          // 获取用户信息
          await store.dispatch('GetInfo')

          isRelogin.show = false
          const accessRoutes = await store.dispatch('GenerateRoutes')

          // 根据roles权限生成可访问的路由表
          accessRoutes.forEach((route) => {
            if (!isHttp(route.path)) {
              try {
                router.addRoute(route) // 动态添加可访问路由表
              } catch (error) {
                console.log(error)
              }
            }
          })
          next({ ...to, replace: true }) // hack方法 确保addRoutes已完成
        } catch (error) {
          store.dispatch('LogOut').then(() => {
            ElMessage.error(error.name)
            setTimeout(() => {
              location.reload()
            }, 1000)
          })
        }
      } else {
        next()
      }
    }
  } else {
    // 没有token
    if (whiteList.indexOf(to.path) !== -1) {
      // 在免登录白名单，直接进入
      next()
    } else {
      next(`/login?redirect=${to.fullPath}`) // 否则全部重定向到登录页
      NProgress.done()
    }
  }
})

router.afterEach(() => {
  NProgress.done()
})
