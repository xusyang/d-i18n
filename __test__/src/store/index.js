import { createStore } from 'vuex'
import app from './modules/app'
import user from './modules/user'
import tagsView from './modules/tagsView'
import permission from './modules/permission'
import settings from './modules/settings'
import base from './modules/base'
import getters from './getters'

const store = createStore({
  modules: {
    app,
    user,
    tagsView,
    permission,
    settings,
    base,
  },
  getters,
})

export default store
