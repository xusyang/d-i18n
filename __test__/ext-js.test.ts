import fse from 'fs-extra'
import path from 'path'
import { describe, expect, it } from 'vitest'
import { di18n } from '../src'
import { TraverseOptions } from '../src/types'
import { filePathWithResult, readFile, writeFile } from '../src/utils'

const traverseOptions: TraverseOptions = {
  forwordOrReverse: true,
  fileType: 'javascript'
}

describe('ext js, file', async () => {
  const dir = './js/'

  const files = await fse
    .readdirSync(path.join(__dirname, dir))
    .filter(x => !x.includes('result'))

  for (let i = 0; i < files.length; i++) {
    const file = files[i]
    it(file, async () => {
      const code = await readFile(dir + file, __dirname)
      let result = di18n(code, traverseOptions)
      writeFile(filePathWithResult(dir + file), result, __dirname)
      expect(result).toMatchSnapshot()
    })
  }
})
