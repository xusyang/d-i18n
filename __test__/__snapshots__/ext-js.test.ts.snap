// Vitest Snapshot v1

exports[`ext js, file > 001. 普通字符串.js 1`] = `
"let province = I18N.$fanyi('重庆')
let str = I18N.$fanyi('中国') + province
"
`;

exports[`ext js, file > 002. 普通字符串-表达式.js 1`] = `
"let str = I18N.$fanyi('中国') + I18N.$fanyi('重庆')
"
`;

exports[`ext js, file > 003. 模板字符串-不带参数.js 1`] = `
"let str = I18N.$fanyi('中国')
"
`;

exports[`ext js, file > 004. 模板字符串-带参数.js 1`] = `
"let country = I18N.$fanyi('中国')
let str = I18N.$fanyi('中国 ') + country
"
`;

exports[`ext js, file > 005. 模板字符串-带多个参数.js 1`] = `
"let country = I18N.$fanyi('中国')
let province = I18N.$fanyi('重庆')
let str = I18N.$fanyi('中国 ') + province
"
`;

exports[`ext js, file > 006. 字符串作为KEY.js 1`] = `
"let obj = {}
obj['我的祖国'] = 1
obj = {
  我的祖国: 1
}
obj['我的' + '祖国'] = 1
"
`;

exports[`ext js, file > 007. 模板字符串作为KEY.js 1`] = `
"let obj = {}
obj[\`我的祖国\`] = 1
obj[\`我的\` + \`祖国\`] = 1
"
`;

exports[`ext js, file > 008. 注释.js 1`] = `
"const a =
  /* this is inner block comment @d-i18n-disable */
  '111'
/* this is block comment */

const b = I18N.$fanyi('222')

// this is line comment

const c = I18N.$fanyi('333')

// this is function comment  @d-i18n-disable

function add() {
  const str = '444'
  console.log(str)
}

// this is arrow function comment

const minus = () => {
  const str = I18N.$fanyi('555')
  console.log(str)
  /**
   * 1
   * 2
   * 3
   */

  // this is block statement comment

  {
    console.log(I18N.$fanyi('666'))
  }
}
"
`;
